docs/querying/aggregations.md

```diff
@@ -238,7 +238,7 @@ Note that queries with first/last aggregators on a segment created with rollup e
 
 (Double/Float/Long/String) ANY aggregator cannot be used in ingestion spec, and should only be specified as part of queries.
 
-If `druid.generic.useDefaultValueForNull=true` aggregation can returns the default value for null and does not prefer "non-null" values over the default value for null. If `druid.generic.useDefaultValueForNull=false`, then aggregation will returns any non-null value.
+If `druid.generic.useDefaultValueForNull=true` aggregation can returns the default value for null and does not prefer "non-null" values over the default value for null. If `druid.generic.useDefaultValueForNull=false`, then aggregation will returns any value including null.
```

Comment:
I think we should re-write this to explain why someone would use this aggregator similar to how it's explained in the snowflake docs - https://docs.snowflake.net/manuals/sql-reference/functions/any_value.html
I'm not sure where the correct place in the docs is to explain this - since technically this is the spec for the native query and we have another page with a spec for sql.

Here's my suggestion for the why:

```
ANY aggregator can be used to simplify and optimize the performance of GROUP BY statements. A common problem for many queries is that the result of a query with a GROUP BY clause can only contain expressions used in the GROUP BY clause itself, or results of aggregate functions

select customer.id , customer.name , sum(orders.value)
    from customer
    join orders on customer.id = orders.customer_id
    group by customer.id , customer.name;

Since we know that customer.id can have only one name, this can be optimized as

select customer.id , ANY(customer.name) , sum(orders.value)
    from customer
    join orders on customer.id = orders.customer_id
    group by customer.id , customer.name;
```

I should also point out, with the current implementation of aggregators, there is no advantage to using an ANY aggregator vs a min aggregator, but maybe that will change in the future 🤷‍♂

Reply:
Suggested re-wording to explain how null handling works

```
This aggregator will return any value for the provided expression. It does not prefer non-null values over null values. If `druid.generic.useDefaultValueForNull` is set to true, this will not return null, but instead return the default value. Otherwise this aggregator may return null
```

Reply:
EDIT: @maytasm3 pointed out this benchmark - https://static.imply.io/gianm/vb.html which shows the performance gain of not having to read a value. Since ANY only reads the first value, this will be faster than min, even thought the aggregator has to loop over all values. 

I should have known that... 🤦‍♂

Reply:
+1 to Suneet's rewording. Agreed as well that we should add the "why". Have filed https://implydata.atlassian.net/browse/DOCS-30 to track.

Reply:
Added the why (short version of what Suneet wrote)